google/cloud/spanner/internal/session_pool.h

```diff
@@ -55,16 +55,21 @@ namespace internal {
 class SessionPool : public std::enable_shared_from_this<SessionPool> {
  public:
   /**
-   * Create a `SessionPool`.
+   * Construct a `SessionPool`.
    *
-   * The parameters allow the `SessionPool` to make remote calls needed to
-   * manage the pool, and to associate `Session`s with the stubs used to
-   * create them. `stubs` must not be empty.
+   * @warning callers should use the `MakeSessionPool()` factory method rather
+   * than calling the constructor and `Initialize()` directly.
    */
   SessionPool(Database db, std::vector<std::shared_ptr<SpannerStub>> stubs,
               SessionPoolOptions options,
               std::unique_ptr<RetryPolicy> retry_policy,
               std::unique_ptr<BackoffPolicy> backoff_policy);
+  /**
+   * If using the factory method is not possible, call `Initialize()` exactly
```

Comment:
I guess the question behind my previous comment is, when might it not be possible to use the factory?

Reply:
It should always be possible to use the factory.